### ISSUE TITLE
getjumplist() return issue

### ISSUE THREAD

**Issue by epheien** [vim/vim#3942]
function f_getjumplist in src/evalfunc.c
```c
    if (list_append_list(rettv->vval.v_list, l) == FAIL)
	return;
    list_append_number(rettv->vval.v_list, (varnumber_T)wp->w_jumplistidx);

    cleanup_jumplist(wp, TRUE);
```
cause getjumplist()[1] not match len(getjumplist()[1]) because cleanup_jumplist() will change wp->w_jumplistidx

**Comment by vim-ml** [comment]
Hi,

On Tue, Feb 12, 2019 at 3:55 AM Epheien <vim-dev-github@256bit.org> wrote:
>
> function f_getjumplist in src/evalfunc.c
>
>     if (list_append_list(rettv->vval.v_list, l) == FAIL)
> return;
>     list_append_number(rettv->vval.v_list, (varnumber_T)wp->w_jumplistidx);
>
>     cleanup_jumplist(wp, TRUE);
>
> cause getjumplist()[1] not match len(getjumplist()[1]) because cleanup_jumplist() will change wp->w_jumplistidx
>

Are you calling the getjumplist() function twice and seeing the returned
w_jumplistidx in the second call is different? or with a single a call to
getjumplist() are you seeing that the returned w_jumplistidx is not correct?

We can fix this by moving the call to cleanup_jumplist() before setting
the w_jumplistidx.

- Yegappan

**Comment by epheien** [comment]
add this to vimrc
```
autocmd VimEnter * echomsg getjumplist()[1] | echomsg getjumplist()[1]
```
and output on my macOS
```
100
16
```

**Comment by vim-ml** [comment]
Hi,

On Tue, Feb 12, 2019 at 7:11 AM Epheien <vim-dev-github@256bit.org> wrote:
>
> add this to vimrc
>
> autocmd VimEnter * echomsg getjumplist()[1] | echomsg getjumplist()[1]
>
> and output on my macOS
>
> 100
> 16
>

Can you try the attached patch?

Thanks,
Yegappan

**Comment by epheien** [comment]
> Hi,
> On Tue, Feb 12, 2019 at 7:11 AM Epheien ***@***.***> wrote: add this to vimrc autocmd VimEnter * echomsg getjumplist()[1] | echomsg getjumplist()[1] and output on my macOS 100 16
> Can you try the attached patch? Thanks, Yegappan

I read this post in github, I can not find any attached patch.
Above is some reproduce issue code.
I think this bug can fix simply by moving the call to cleanup_jumplist() before setting the w_jumplistidx.

You can fix and close this issue.

**Comment by brammool** [comment]
Fixed by patch 8.1.0901
